crates/proof-of-sql/src/sql/parse/query_context_builder.rs

```
@@ -110,17 +107,21 @@ impl<'a> QueryContextBuilder<'a> {
         clippy::missing_panics_doc,
         reason = "The assertion ensures there is at least one column, and this is a fundamental requirement for schema retrieval."
     )]
-    fn lookup_schema(&self) -> Vec<(Identifier, ColumnType)> {
+    fn lookup_schema(&self) -> Vec<(Ident, ColumnType)> {
         let table_ref = self.context.get_table_ref();
         let columns = self.schema_accessor.lookup_schema(*table_ref);
         assert!(!columns.is_empty(), "At least one column must exist");
         columns
     }
 
+    /// # Panics
+    /// This function will panic if the conversion from `Ident` to `Identifier` fails.
     fn visit_select_all_expr(&mut self) -> ConversionResult<()> {
         for (column_name, _) in self.lookup_schema() {
-            let col_expr = Expression::Column(column_name);
-            self.visit_aliased_expr(AliasedResultExpr::new(col_expr, column_name))?;
+            let column_identifier =
+                Identifier::try_from(column_name).expect("Failed to convert Ident to Identifier");
```

This is still potentially panicking, rather than returning a `Result`.